pinot-common/src/main/java/org/apache/pinot/common/utils/config/InstanceUtils.java

```diff
@@ -119,4 +122,36 @@ public static void updateHelixInstanceConfig(InstanceConfig instanceConfig, Inst
       mapFields.remove(POOL_KEY);
     }
   }
+
+  /**
+   * Returns an instance admin URL from a given InstanceConfig. Will decipher whether the URL protocol should
+   * be HTTP vs HTTPS.
+   */
+  public static URL getInstanceURL(InstanceConfig instanceConfig) {
```

Comment:
This is for pinot controller only, for pinot broker, you will have one port for query and one port for admin.

Reply:
Maybe change this method to getInstanceAdminURL

Reply:
Going to go the approach of leveraging the ExtraInstanceConfig class.